elpis/wrappers/objects/pron_dict.py

```diff
@@ -0,0 +1,81 @@
+import json
+import shutil
+import glob
+import os
+import threading
+
+from pathlib import Path
+from io import BufferedIOBase
+from multiprocessing.dummy import Pool
+from shutil import move
+
+from elpis.wrappers.objects.dataset import Dataset
+from elpis.wrappers.objects.fsobject import FSObject
+from elpis.wrappers.objects.path_structure import existing_attributes, ensure_paths_exist
+from elpis.wrappers.input.make_prn_dict import generate_pronunciation_dictionary
+
+
+class PronDict(FSObject):
+
+    # The configuration settings stored in the file below.
+    _config_file = 'pron_dict.json'
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.dataset: Dataset
+        self.dataset = None
+        self.config['dataset_name'] = None  # dataset hash has not been linked
+        self.l2s_path = self.path.joinpath('l2s.txt')
+        self.lexicon_txt = self.path.joinpath('lexicon.txt')
+        self.config['l2s'] = None  # file has not been uploaded
+
+    @classmethod
+    def load(cls, base_path: Path):
+        self = super().load(base_path)
+        self.l2s_path = self.path.joinpath('l2s.txt')
+        self.lexicon_txt = self.path.joinpath('lexicon.txt')
+        self.dataset = None
+        return self
+
+    def link(self, dataset: Dataset):
+        self.dataset = dataset
+        self.config['dataset_name'] = dataset.name
+
+    def set_l2s_path(self, path: Path):
+        path = Path(path)
+        with path.open(mode='rb') as fin:
+            self.set_l2s_fp(fin)
+
+    def set_l2s_fp(self, file: BufferedIOBase):
+        self.set_l2s_content(file.read())
+
+    def set_l2s_content(self, content: str):
+        # TODO: this function uses parameter str, and must be bytes or UTF-16
```

Comment:
Not sure how to check for UTF-16, but if you need to support multiple input types:

```
from typing import Union

def set_l2s_content(self, content: Union[str, bytes]) -> None:
```

Reply:
Ah, will investigate. Thanks for the heads-up